### PR TITLE
Change GenerateMethod return value to TextEdits

### PR DESCRIPTION
**Pull request by BladeMF**
Changed the refactoring's return value. Once we agree on this, I will change the rest in order to prepare them to be used in LSP. Haven't done the **phpactor** change as well (`TextEdits->apply(SourceCode)`).

Also, for some reason I had to add `allowRisky` for the CS fixer, because it would run otherwise.

### CODE REVIEW

File: tests/Adapter/WorseReflection/Refactor/WorseGenerateMethodTest.php [hunk 1]
```diff
@@ -89,11 +89,17 @@ public function greet()
...
         $factory = new WorseBuilderFactory($reflector);
         $generateMethod = new WorseGenerateMethod($reflector, $factory, $this->updater());
-        return $generateMethod->generateMethod(SourceCode::fromString($source), $start, $name);
+        $sourceCode = SourceCode::fromString($source);
+        $edits = $generateMethod->generateMethod($sourceCode, $start, $name);
+        $transformed = SourceCode::fromStringAndPath(
+            (string) $edits->apply($sourceCode),
+            $sourceCode->path()
+        );
+        return $transformed;
```
Comment (dantleech) [inline tests/Adapter/WorseReflection/Refactor/WorseGenerateMethodTest.php:106]: not important, but could inline some of this and remove the temporary variables.

**Comment by dantleech** [comment]
:+1:

**Comment by BladeMF** [comment]
Don't merge it, it doesn't work! The tests don't test how it handles changes in other documents.

File: lib/Adapter/WorseReflection/Refactor/WorseGenerateMethod.php [hunk 2]
```diff
@@ -57,10 +58,7 @@ public function generateMethod(SourceCode $sourceCode, int $offset, ?string $met
         $prototype = $this->addMethodCallToBuilder($methodCall, $visibility, $methodCall->isStatic(), $methodName);
         $sourceCode = $this->resolveSourceCode($sourceCode, $methodCall, $visibility);
 
-        return SourceCode::fromStringAndPath(
-            (string) $this->updater->textEditsFor($prototype, Code::fromString((string) $sourceCode))->apply($sourceCode),
-            $sourceCode->path()
```
Comment (dantleech) [inline lib/Adapter/WorseReflection/Refactor/WorseGenerateMethod.php:62]: we need to add something like `TextDocumentEdits` value object to the text-document package and return that or an additional `WorkspaceEdits` which is a collection of `TextDocumentEdits`

Comment (BladeMF) [reply lib/Adapter/WorseReflection/Refactor/WorseGenerateMethod.php:62]: Ahahaha, just noticed the same thing. Posted a discussion [here](https://github.com/phpactor/phpactor/discussions/1252).

**Comment by dantleech** [comment]
the text edits dep needs to be bumped ..

**Comment by BladeMF** [comment]
> the text edits dep needs to be bumped ..

Will do.

**Pull request merged**